src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -132,15 +172,40 @@ const useGroupInfo = (channelId = null) => {
   const uploadImage = async (pathImg) => {
     try {
       setIsUploadingImage(true);
+
+      setUploadedImage(pathImg);
+      const channelListTemp = await ChannelListSchema.getSchemaById(localDb, channelId);
+      channelListTemp.channelPicture = pathImg;
+      await channelListTemp.save(localDb);
+      setSelectedChannel(channelListTemp);
+
+      refresh('channelList');
+      refreshWithId('chat', channelId);
+      refresh('channelInfo');
+
       const result = await ImageUtils.uploadImage(pathImg);
       setUploadedImage(result.data.url);
-      const dataEdit = {
-        name: chatName,
-        image: result.data.url
-      };
 
-      await channel.update(dataEdit);
       setIsUploadingImage(false);
+
+      const responseChannelData = await SignedMessageRepo.changeSignedChannelDetail(
+        channelId,
+        null,
+        result.data.url
+      );
+      const channelList = await ChannelListSchema.getSchemaById(localDb, channelId);
+      channelList.channelPicture = result.data.url;
+      channelList.rawJson = responseChannelData;
+      await channelList.save(localDb);
+      setSelectedChannel(channelList);
+
+      refresh('channelList');
+      refreshWithId('chat', channelId);
+      refresh('channelInfo');
+
+      setTimeout(() => {
+        navigation.navigate('SignedChatScreen');
+      }, 500);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `uploadImage` function has been updated to handle image uploading and updating the channel details with the new image. It also updates the local database and refreshes the UI accordingly. Error handling is done properly with a try-catch block. However, it's recommended to provide user feedback in case of an error instead of just logging it.
```diff
+ const uploadImage = async (pathImg) => {
+   // ... rest of the code
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->